generators/jsgenerator.coffee

```diff
@@ -83,23 +86,27 @@ generator =
     emit '}', whileBuffer
     whileBuffer.join '\n'
 
-  ReturnStatement: (s) -> emit "return #{gen s.value};"
+  ReturnStatement: (s) ->
```

Comment:
Rather than having the same logic in 5 or 6 different places (because we might not always have a block that needs to be returned, I.E. a single line if statement), we do this hack so that a return statement only outputs a return when it's valid.

It's a bit less DRY, but certainly confusing.